contracts/test/integration/IntegrationTestMorphoCompoundPCVDeposit.t.sol

```diff
@@ -0,0 +1,122 @@
+import {Core} from "../../core/Core.sol";
+import {IVolt} from "../../volt/IVolt.sol";
+import {DSTest} from "../unit/utils/DSTest.sol";
+import {IDSSPSM} from "../../pcv/maker/IDSSPSM.sol";
+import {Constants} from "../../Constants.sol";
+import {PCVGuardian} from "../../pcv/PCVGuardian.sol";
+import {MainnetAddresses} from "./fixtures/MainnetAddresses.sol";
+import {PegStabilityModule} from "../../peg/PegStabilityModule.sol";
+import {ERC20CompoundPCVDeposit} from "../../pcv/compound/ERC20CompoundPCVDeposit.sol";
+import {MorphoCompoundPCVDeposit} from "../../pcv/morpho/MorphoCompoundPCVDeposit.sol";
+
+import "hardhat/console.sol";
+
+contract IntegrationTestMorphoCompoundPCVDeposit is DSTest {
+    using SafeCast for *;
+
+    Vm public constant vm = Vm(HEVM_ADDRESS);
+
+    MorphoCompoundPCVDeposit private daiDeposit;
+    MorphoCompoundPCVDeposit private usdcDeposit;
+
+    PCVGuardian private immutable pcvGuardian =
+        PCVGuardian(MainnetAddresses.PCV_GUARDIAN);
+
+    Core private core = Core(MainnetAddresses.CORE);
+    PegStabilityModule private daiPSM =
+        PegStabilityModule(MainnetAddresses.VOLT_DAI_PSM);
+
+    IERC20 private dai = IERC20(MainnetAddresses.DAI);
+    IERC20 private usdc = IERC20(MainnetAddresses.USDC);
+
+    uint256 public daiBalance;
+    uint256 public usdcBalance;
+
+    uint256 targetDaiBalance = 100_000e18;
+    uint256 targetUsdcBalance = 100_000e6;
+
+    function setUp() public {
+        daiDeposit = new MorphoCompoundPCVDeposit(
+            address(core),
+            MainnetAddresses.CDAI
+        );
+        usdcDeposit = new MorphoCompoundPCVDeposit(
+            address(core),
+            MainnetAddresses.CUSDC
+        );
+
+        vm.startPrank(MainnetAddresses.DAI_USDC_USDT_CURVE_POOL);
+        dai.transfer(address(daiDeposit), targetDaiBalance);
+        usdc.transfer(address(usdcDeposit), targetUsdcBalance);
+        vm.stopPrank();
+
+        usdcDeposit.deposit();
+        daiDeposit.deposit();
+    }
+
+    function testSetup() public {
+        assertEq(address(daiDeposit.core()), address(core));
+        assertEq(address(usdcDeposit.core()), address(core));
+
+        assertEq(daiDeposit.balanceReportedIn(), address(dai));
+        assertEq(usdcDeposit.balanceReportedIn(), address(usdc));
+
+        assertEq(address(daiDeposit.cToken()), address(MainnetAddresses.CDAI));
+        assertEq(
+            address(usdcDeposit.cToken()),
+            address(MainnetAddresses.CUSDC)
+        );
+
+        assertEq(address(daiDeposit.token()), address(MainnetAddresses.DAI));
+        assertEq(address(usdcDeposit.token()), address(MainnetAddresses.USDC));
+
+        assertApproxEq(
+            daiDeposit.balance().toInt256(),
+            targetDaiBalance.toInt256(),
+            0
+        );
+        assertApproxEq(
+            usdcDeposit.balance().toInt256(),
+            targetUsdcBalance.toInt256(),
+            0
+        );
+    }
+
+    function testWithdraw() public {
```

Comment:
add fuzz test
add test that does partial withdraw
assert that morpho rewards accrue correctly